### PR TITLE
Add tablet type filtering to vtctldclient GetTablets

### PR DESCRIPTION
**Pull request by mattlord**
## Description

Filtering by tablet type was added to the `ListAllTablets` vtctl command in https://github.com/vitessio/vitess/pull/9560 but did not exist in the new `GetTablets` vtctldclient command (its successor).

## Related Issue(s)

  - Follow-up to: https://github.com/vitessio/vitess/pull/9560

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation: https://github.com/vitessio/website/pull/1446

### CODE REVIEW

**Review by notfelineit — APPROVED** [review]
LGTM! 😄 prob should wait for @ajm188 's review

File: go/cmd/vtctldclient/command/tablets.go [hunk 1]
```diff
@@ -421,11 +427,20 @@ func commandGetTablets(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("--shard (= %s) cannot be passed without also passing --keyspace", getTabletsOptions.Shard)
 	}
 
+	tabletType := topodatapb.TabletType_UNKNOWN // Zero Value
```
Comment (ajm188) [inline go/cmd/vtctldclient/command/tablets.go:430]: there is a tablet-type flag you can use in `topoproto` to parse directly into the correct type

File: go/cmd/vtctldclient/command/tablets.go [hunk 2]
```diff
@@ -99,7 +99,7 @@ Note: hook names may not contain slash (/) characters.
 	}
 	// GetTablets makes a GetTablets gRPC call to a vtctld.
 	GetTablets = &cobra.Command{
-		Use:   "GetTablets [--strict] [{--cell $c1 [--cell $c2 ...], --keyspace $ks [--shard $shard], --tablet-alias $alias}]",
+		Use:   "GetTablets [--strict] [{--cell $c1 [--cell $c2 ...], [--tablet-type $t1] $ks [--shard $shard], --tablet-alias $alias}]",
```
Comment (ajm188) [inline go/cmd/vtctldclient/command/tablets.go:102]: just realized this is wrong, `--keyspace` is still a flag argument if i'm reading the rest right

File: go/cmd/vtctldclient/command/tablets.go [hunk 3]
```diff
@@ -634,6 +641,8 @@ func init() {
 
 	GetTablets.Flags().StringSliceVarP(&getTabletsOptions.TabletAliasStrings, "tablet-alias", "t", nil, "List of tablet aliases to filter by.")
 	GetTablets.Flags().StringSliceVarP(&getTabletsOptions.Cells, "cell", "c", nil, "List of cells to filter tablets by.")
+	GetTablets.Flags().Var((*topoproto.TabletTypeFlag)(&getTabletsOptions.TabletType), "tablet-type", fmt.Sprintf("Tablet type to filter by (%s).",
```
Comment (ajm188) [inline go/cmd/vtctldclient/command/tablets.go:644]: why include the full list here? that might be better placed in the `Long` text

File: go/vt/topo/topoproto/tablet.go [hunk 4]
```diff
@@ -227,6 +227,25 @@ func MakeStringTypeList(types []topodatapb.TabletType) []string {
...
 }
 
+// MakeUniqueStringTypeList returns a unique list of strings that match
+// the input list -- with duplicate types removed.
+// This is needed as some types are aliases for others, like BATCH and
+// RDONLY, so e.g. rdonly shows up twice in the list when using
+// AllTabletTypes.
+func MakeUniqueStringTypeList(types []topodatapb.TabletType) []string {
+	strs := make([]string, 0, len(types))
+	seen := make(map[string]bool)
```
Comment (ajm188) [inline go/vt/topo/topoproto/tablet.go:237]: if we end up keeping this, `map[string]struct{}` is better (empty struct is zero bytes whereas bool is 1)

**Review by ajm188 — COMMENTED** [review]
couple small things

**Review by ajm188 — APPROVED** [review]

**Pull request merged**